orchestration/src/main/java/com/sap/ai/sdk/orchestration/OrchestrationChatResponse.java

```diff
@@ -24,18 +28,46 @@ public class OrchestrationChatResponse {
    */
   @Nonnull
   public String getContent() throws OrchestrationClientException {
-    final var choices =
-        ((LLMModuleResultSynchronous) originalResponse.getOrchestrationResult()).getChoices();
-
-    if (choices.isEmpty()) {
-      return "";
-    }
-
-    final var choice = choices.get(0);
+    final var choice = getCurrentChoice();
 
     if ("content_filter".equals(choice.getFinishReason())) {
       throw new OrchestrationClientException("Content filter filtered the output.");
     }
     return choice.getMessage().getContent();
   }
+
+  /**
+   * Get the token usage.
+   *
+   * @return The token usage.
+   */
+  @Nonnull
+  public TokenUsage getTokenUsage() {
+    return ((LLMModuleResultSynchronous) originalResponse.getOrchestrationResult()).getUsage();
+  }
+
+  /**
+   * Get all messages. This can be used for subsequent prompts as a message history.
+   *
+   * @return A list of all messages.
+   */
+  @Nonnull
+  public List<ChatMessage> getAllMessages() {
+    final var messages = originalResponse.getModuleResults().getTemplating();
```

Comment:
```suggestion
    final var messages = new ArrayList<>(originalResponse.getModuleResults().getTemplating());
```

Otherwise, the subsequent `.add` will manipulate the original response. Repeated calls to `getAllMessages()` thus start to contain the assistant message multiple times. Unfortunately, this is currently missed by the test assertions. Consider adding `assertThat(result.getAllMessages()).hasSize(2);` in the unit test below